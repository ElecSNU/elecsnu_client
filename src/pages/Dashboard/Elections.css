.polls {
    margin: 1.5rem 0;
    padding: 1.5% 3%;
}

.poll {
    color: var(--dark);
    padding: 2%;
    margin: 1.5% 0;
    cursor: pointer;
    border: 2.5px solid rgba(0, 0, 0, 0);
    transition: all 0.4s;
}

.poll:hover {
    border: 2.5px solid var(--accent1);
    color: var(--accent1);
}
