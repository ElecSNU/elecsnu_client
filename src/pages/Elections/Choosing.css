#choosing {
    height: 100%;
    padding: 5%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

#choosing > * {
    width: 100%;
}

.candidate {
    padding: 2.5%;
    margin: 2.5%;
    transition: all 0.7s;
    border: 5px solid transparent;
    cursor: pointer;
}

.candidate h1 {
    transition: all 0.7s;
}

.candidate:hover,
.candidate.chosen {
    border: 5px solid var(--accent1);
}

.candidate:hover h1,
.candidate.chosen h1 {
    color: var(--accent1);
}

#choosing .submit-button {
    width: 50%;
}
