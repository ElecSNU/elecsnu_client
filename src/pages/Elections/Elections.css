#election {
    display: flex;
    height: 75%;
    width: 85%;
    padding: 2.5% 3.5%;
    overflow: hidden;
}

#election-content {
    width: 100%;
    justify-content: space-between;
}

.election-heading {
    width: 100%;
    height: auto;
    text-align: center;
    padding: 1.5% 0;
}

.candidate-container {
    margin: 1% 0;
    overflow: hidden;
    overflow-y: scroll;
}

.candidates {
    margin-top: 1.5%;
}

.candidate {
    padding: 1% 2.5%;
    margin: 1.5% 0;
    align-items: center;
    justify-content: space-between;
}

.candidate a {
    width: 25%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    transition: color 0.7s;
    cursor: pointer;
}

.candidate a:hover {
    color: var(--accent1);
}

.candidate a img {
    height: 100%;
    width: 10%;
}

#election .submit-button {
    align-self: center;
    background-color: var(--accent2);
}

#election .submit-button.disabled {
    filter: grayscale();
    pointer-events: none;
}
