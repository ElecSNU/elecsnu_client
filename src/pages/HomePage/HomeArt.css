#home {
    height: 100%;
    display: flex;
    align-items: center;
    overflow: hidden !important;
}

#home .voting-banner {
    margin: 0 5rem;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

#home .voting-banner .submit-button {
    margin-top: 2rem;
    width: 100%;
}

#home .home-art {
    width: 100%;
    height: 100%;
    position: relative;
    overflow: hidden;
}

#home .home-art img {
    width: 120%;
    position: absolute;
    right: -25%;
    bottom: 0;
}
