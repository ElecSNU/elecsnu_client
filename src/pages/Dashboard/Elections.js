import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';

import './Elections.css';

const Elections = ({ polls, state }) => {
    const [startPoll, setStartPoll] = useState(null);

    return startPoll != null ? (
        <Redirect to={'/dashboard/poll/' + startPoll} />
    ) : (
        <div>
            <h1 className='heading-text'>
                {state} Elections
            </h1>
            <div
                className={`polls ${
                    state === 'active'
                        ? 'background-1'
                        : 'background-accent2'
                } border-radius-15`}
            >
                {polls.map((poll, index) => (
                    <div
                        className='poll border-radius-15 background-light subheading-text'
                        onClick={() => {
                            setStartPoll(poll.id);
                        }}
                        key={index}
                    >
                        {poll.election_name}
                    </div>
                ))}
            </div>
        </div>
    );
};

export default Elections;
