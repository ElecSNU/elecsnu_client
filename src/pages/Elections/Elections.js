import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Link } from 'react-router-dom';
import useFirestore from '../../hooks/useFirestore';

import Manifesto_icon from '../../assets/images/manifesto_icon.svg';

import './Elections.css';

const Elections = () => {
    const params = useParams();
    const electionDocs = useFirestore('newelections').docs;

    const [electionDetails, setElectionDetails] = useState(
        {}
    );

    const [totalCandidates, setTotalCandidates] = useState(
        0
    );

    // 0 ==> Active
    // 1 ==> Past
    // 2 ==> Upcoming
    const [electionState, setElectionState] = useState(2);

    const getElectionDetails = () => {
        const userRoll = window.localStorage.getItem(
            'user_roll'
        );
        electionDocs.forEach((election) => {
            if (election.id === params.election_id) {
                let candidateNames = [];
                election.election_candidates.forEach(
                    (c) => {
                        candidateNames = [
                            ...candidateNames,
                            c.candidate_name,
                        ];
                    }
                );

                setTotalCandidates(candidateNames);

                setElectionDetails(election);
                if (!election.started) {
                    setElectionState(2);
                } else if (
                    election.started &&
                    !election.ended &&
                    !election.voters.includes(
                        Number(userRoll)
                    )
                ) {
                    setElectionState(0);
                } else {
                    setElectionState(1);
                }
                return;
            }
        });
    };

    useEffect(() => {
        getElectionDetails();

        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [electionDocs]);

    return (
        <div
            id='election'
            className='border-radius-15 box-shadow background-light-translucent'
        >
            <div
                id='election-content'
                className='d-flex-column'
            >
                <div className='election-heading heading-text background-accent1 border-radius-15 foreground-light'>
                    Voting - {electionDetails.election_name}
                </div>
                <div className='candidate-container'>
                    <div className='heading-text foreground-accent2'>
                        CANDIDATE LIST
                    </div>
                    <div className='candidates '>
                        {electionDetails !== {} &&
                            electionDetails.election_candidates &&
                            electionDetails.election_candidates.map(
                                (candidate) => {
                                    return (
                                        <div
                                            className='candidate d-flex background-1 border-radius-15'
                                            key={
                                                candidate.candidate_roll_no
                                            }
                                        >
                                            <h2 className='candidate-name heading-text capital-text foreground-light'>
                                                {
                                                    candidate.candidate_name
                                                }
                                            </h2>
                                            <a
                                                href={
                                                    candidate.candidate_manifesto
                                                }
                                                target='__blank'
                                                className='candidate-manifesto heading-text foreground-light'
                                            >
                                                MANIFESTO
                                                <img
                                                    src={
                                                        Manifesto_icon
                                                    }
                                                    alt='.'
                                                />
                                            </a>
                                        </div>
                                    );
                                }
                            )}
                    </div>
                </div>
                {electionState === 1 ? (
                    <Link
                        className='submit-button background-accent1'
                        onClick={() => {}}
                        to={`/dashboard/results/${params.election_id}`}
                    >
                        RESULTS
                    </Link>
                ) : (
                    <Link
                        className={`submit-button background-accent1 ${
                            electionState === 0
                                ? ''
                                : 'disabled'
                        }`}
                        onClick={() =>
                            console.log('Clicked')
                        }
                        to={{
                            pathname: `/dashboard/poll/${params.election_id}/voting`,
                            search: `?candidates=${totalCandidates.toString()}`,
                        }}
                    >
                        VOTE
                    </Link>
                )}
            </div>
        </div>
    );
};

export default Elections;
