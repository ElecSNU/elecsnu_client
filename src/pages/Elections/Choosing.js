import React, { useState } from 'react';
import useFirestore from '../../hooks/useFirestore';
import { fire_store } from '../../configs/firebase/config';
import { toast } from 'react-toastify';
import useLoader from '../../hooks/useLoader';

import Share from './Share';

import './Choosing.css';

const Choosing = ({ candidates }) => {
    const [chosen, setChosen] = useState(0);
    const electionDocs = useFirestore('newelections').docs;
    const showLoader = useLoader();
    const [done, setDone] = useState(false);
    const [electionName, setElectionName] = useState('');

    const castVote = () => {
        showLoader(true);
        const electionId = window.location.pathname.split(
            '/'
        )[3];
        let electionDetails = {};
        electionDocs.forEach((e) => {
            if (e.id === electionId) {
                electionDetails = e;
                return;
            }
        });
        setElectionName(electionDetails.election_name);

        let userGender = window.localStorage.getItem(
            'voter_gender'
        );
        let userDept = window.localStorage.getItem(
            'voter_dept'
        );
        let userBatch = window.localStorage.getItem(
            'voter_batch'
        );
        let userRoll = window.localStorage.getItem(
            'user_roll'
        );

        electionDetails.voters = [
            ...electionDetails.voters,
            Number(userRoll),
        ];
        electionDetails.election_votes = [
            ...electionDetails.election_votes,
            {
                voter_gender: userGender,
                groups: `${userDept},Batch${userBatch}`,
                candidate_chosen: chosen,
            },
        ];

        fire_store
            .collection('newelections')
            .doc(electionId)
            .set(electionDetails);

        showLoader(false);

        toast.success('Successfully casted vote!');

        setDone(true);
    };

    return (
        <>
            {done ? (
                <Share electionName={electionName} />
            ) : (
                <></>
            )}
            <div id='choosing'>
                <div className='heading-text capital-text center-text foreground-back'>
                    Who would you want to vote for?
                </div>
                <div className='candidates'>
                    {candidates.map((candidate, index) => {
                        return (
                            <div
                                className={`candidate background-1 border-radius-15 ${
                                    index === chosen
                                        ? 'chosen'
                                        : ''
                                }`}
                                key={index}
                                onClick={() =>
                                    setChosen(index)
                                }
                            >
                                <h1 className='capital-text subheading-text foreground-light center-text'>
                                    {candidate}
                                </h1>
                            </div>
                        );
                    })}
                </div>
                <button
                    className='submit-button'
                    onClick={castVote}
                >
                    CAST VOTE
                </button>
            </div>
        </>
    );
};

export default Choosing;
