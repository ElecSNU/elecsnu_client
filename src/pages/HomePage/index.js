import React from 'react';

import './HomeArt.css';

import homeArt from '../../assets/images/home-art.svg';
import sc1 from '../../assets/images/scroll_1.png';
import sc2 from '../../assets/images/scroll_2.png';
// import gfont from 'https://fonts.googleapis.com/css?family=Montserrat' 

const HomePage = () => {
   
    return (
        <>
            <div
                id='home'
                style={{
                    width: '100%',
                    height: '100vh',
                    backgroundColor: '#004aad'
                }}
            >
                <div className='voting-banner' >
                        <h1  style={{
                        // fontFamily: "Montserrat"
                    }}>LET'S MAKE VOTING FUN!</h1>
                        <br/>
                        <h4 >
                            A step towards secure and intuitive digital voting 
                        </h4>

                </div>
                <div className='home-art'>
                    <img src={homeArt} alt='home-art' />
                </div>
            </div>
            {/* <div className='sc1'> */}
                <h1>All about online voting systems</h1>     
            {/* </div>s */}
            <div className='sc1'>
                <img src={sc1} alt='sc1' />    
            </div>
            <div className='sc1'>
                <img src={sc2} alt='sc1' />    
            </div>
            
        </>
    );
};

export default HomePage;
