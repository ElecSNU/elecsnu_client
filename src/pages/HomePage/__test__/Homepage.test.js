import { render, screen, fireEvent } from '@testing-library/react';
import Elections from './../index';
import index from './../index';
import ReactDOM from 'react-dom'
import React , { useState }from 'react'
import renderer from 'react-test-renderer'
import HomePage from './../index';


test('Tests the rendering of the page', () => {
    const div = document.createElement("div")
    const tree = renderer.create(<HomePage></HomePage>).toJSON();
    expect(tree).toMatchSnapshot();
   });
   