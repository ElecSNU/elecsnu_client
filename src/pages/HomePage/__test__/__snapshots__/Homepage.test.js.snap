// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of the page 1`] = `
<div
  id="home"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="voting-banner"
  >
    <h1>
      Let's make voting fun!
    </h1>
    <button
      className="submit-button"
    >
      Vote Now!
    </button>
  </div>
  <div
    className="home-art"
  >
    <img
      alt="home-art"
      src="home-art.svg"
    />
  </div>
</div>
`;
